src/core/server/rendering/rendering_service.tsx

```diff
@@ -94,6 +98,21 @@ export class RenderingService {
       user: isAnonymousPage ? {} : await uiSettings.getUserProvided(),
     };
 
+    let clusterInfo = {};
+    try {
+      // Only provide the clusterInfo if the request is authenticated and the elasticsearch service is available.
+      if (isAuthenticated(http.auth, request) && elasticsearch) {
```

Comment:
I wonder if we need a dedicated `isAuthenticated` check, _or_ if we can get away with using the existing `isAnonymousPage` boolean. WDYT?

Reply:
I completely missed that boolean there! Good point!

Reply:
I've looked through the code to figure out when the flag is set. If the user navigates to `/app/home`, they are navigating to a non-anonymous page, so `isAnonymousPage` is `false`. The HTTP router relies on the authentication mechanism to forbid access to that route in the first place, so it should be OK. However, it feels safer to validate it instead of relying on the flag. WDYT?